### DISCUSSION TITLE
Admin edit row Basic users get kicked to random page

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2145] [Help] [answered]
Hi, I have started to implement roles in my application where admin users/power users can see different screens from the other users. 
Is there a way I can make a button do different things based on the user's role? for example, if you are an admin you can edit data by clicking edit if you are not an admin it will either do nothing or take you to a specified page. right now I am experiencing that if you do not have rights to that page then it will kick you to some random page when clicking on the button. 

![image](https://user-images.githubusercontent.com/80058238/127242169-d1e587d0-24d9-4d72-8c49-858d4868b212.png)

**Comment by mjashanks** [comment, marked as answer]
Hi Andrew.

This is possible with the new conditional UI - you'll need to update.

In this video, I create 2 buttons. 
1. View basic - hidden if role is ADMIN
2. View Admin - hidden if role is not ADMIN

The only annoyance is that the `View Basic` is always hidden inside the builder, as the builder is an ADMIN. You can still access it via the component tree on the left.


https://user-images.githubusercontent.com/3524181/127311120-51aa1e17-cd29-498b-a396-651812800a1e.mp4

**Reply by KernOil** [reply]
@mjashanks This is perfect Thank you very much for the video!

**Reply by KernOil** [reply]
@mjashanks  I see the 0.9.84 update fixed the problems I was having with my tables (dates and key row) do you know if this update includes the ability to use option pickers directly to a SQL DB? I know that was on the road map was that part of this release?

**Reply by mjashanks** [reply]
This will come next month!
